tests/controllers/feature/feature_test.go

```diff
@@ -0,0 +1,204 @@
+package feature_test
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/controllers/management/feature"
+	"github.com/rancher/rancher/pkg/features"
+	"github.com/rancher/rancher/pkg/wrangler"
+	"github.com/rancher/wrangler/v2/pkg/crd"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+type FeatureTestSuite struct {
+	suite.Suite
+	ctx             context.Context
+	cancel          context.CancelFunc
+	testEnv         *envtest.Environment
+	wranglerContext *wrangler.Context
+}
+
+var (
+	harvesterFeatureNoAnnotation = v3.Feature{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "harvester-baremetal-container-workload",
+		},
+	}
+	harvesterFeature = v3.Feature{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "harvester-baremetal-container-workload",
+			Annotations: map[string]string{
+				v3.ExperimentalFeatureValue: v3.ExperimentalFeatureKey,
+			},
+		},
+	}
+	//t = true
+	//f = false
+	h = v3.Feature{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "harvester",
+		},
+	}
+)
+
+func (s *FeatureTestSuite) SetupSuite() {
+	s.ctx, s.cancel = context.WithCancel(context.TODO())
+	// Load CRD from YAML for REST Client
+	s.testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{
```

Comment:
This is not needed as we are using `factory.BatchCreateCRDs` to register the CRDs